lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -4,221 +4,494 @@
-        if (value.isNaN)
-          return new ByteData(8)
-            ..setUint32(0, 0x00000000, Endian.little)
-            ..setUint32(4, 0x7ff80000, Endian.little);
-        return new ByteData(8)..setFloat64(0, value, Endian.little);
-      })
-      ..[PbFieldType._FLOAT_BIT] = makeWriter((double value) {
-        const double MIN_FLOAT_DENORM = 1.401298464324817E-45;
-        const double MAX_FLOAT = 3.4028234663852886E38;
-        // TODO(antonm): reevaluate once semantics of odd values
-        // writes is clear.
-        if (value.isNaN) return makeByteData32(0x7fc00000);
-        if (value.abs() < MIN_FLOAT_DENORM) {
-          return makeByteData32(value.isNegative ? 0x80000000 : 0x00000000);
-        }
-        if (value.isInfinite || value.abs() > MAX_FLOAT) {
-          return makeByteData32(value.isNegative ? 0xff800000 : 0x7f800000);
+    // Move the current output chunk into _outputChunks and commit the current
+    // splice for uniformity.
+    _commitChunk(false);
+    _commitSplice();
+
+    int outPos = offset; // Output position in the buffer.
+    int chunkIndex = 0, chunkPos = 0; // Position within _outputChunks.
+    for (int i = 0; i < _splices.length; i++) {
+      final action = _splices[i];
+      if (action is int) {
+        if (action <= 0) {
+          // action is a positive varint to be emitted into the output buffer.
+          int v = -action;
+          while (v >= 0x80) {
+            buffer[outPos++] = 0x80 | (v & 0x7f);
+            v >>= 7;
+          }
+          buffer[outPos++] = v;
+        } else {
+          // action is an amount of bytes to copy from _outputChunks into the
+          // buffer.
+          int bytesToCopy = action;
+          while (bytesToCopy > 0) {
+            final chunk = _outputChunks[chunkIndex];
+            final bytesInChunk = _outputChunks[chunkIndex + 1];
+
+            // Copy at most bytesToCopy bytes from the current chunk.
+            final leftInChunk = bytesInChunk - chunkPos;
+            final bytesToCopyFromChunk =
+                leftInChunk > bytesToCopy ? bytesToCopy : leftInChunk;
+            final endPos = chunkPos + bytesToCopyFromChunk;
+            while (chunkPos < endPos) {
+              buffer[outPos++] = chunk[chunkPos++];
+            }
+            bytesToCopy -= bytesToCopyFromChunk;
+
+            // Move to the next chunk if the current one is exhausted.
+            if (chunkPos == bytesInChunk) {
+              chunkIndex += 2;
+              chunkPos = 0;
+            }
+          }
         }
-        return new ByteData(4)..setFloat32(0, value, Endian.little);
-      })
-      ..[PbFieldType._ENUM_BIT] =
-          makeWriter((value) => _int32ToBytes(value.value))
-      ..[PbFieldType._GROUP_BIT] = (output, value) {
-        value.writeToCodedBufferWriter(output);
+      } else {
+        // action is a TypedData containing bytes to emit into the output
+        // buffer.
+        outPos = _copyInto(buffer, outPos, action);
```

Comment:
The public API is List<int> but _copyInto takes only Uint8List.
Probably best to push the constraint up to the public API, since it is broken is called with a different type of list

Reply:
Done.

Reply:
I tightened the signature and updated CHANGELOG.md with information about the breaking change.